File: add_to_cart_test.py
 

import unittest
from selenium import webdriver

class AddToCartTest(unittest.TestCase):
    def setUp(self):
        self.driver = webdriver.Chrome()
        self.driver.get("https://example.com")  # Replace with the URL of your e-commerce website

    def test_add_to_cart(self):
        # Test case for adding items to the cart
        # Steps:
        # 1. Navigate to the product page.
        # 2. Click on the "Add to Cart" button.
        # 3. Verify that the item is added to the cart.
        product_page_button = self.driver.find_element_by_css_selector("#add-to-cart-button")
        product_page_button.click()
        cart_icon = self.driver.find_element_by_css_selector(".cart-icon")
        self.assertTrue(cart_icon.is_displayed(), "Item not added to cart")

    def tearDown(self):
        self.driver.quit()

if __name__ == "__main__":
    unittest.main()

File: AddToCartTest.java
 import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import static org.junit.Assert.assertTrue;

public class AddToCartTest {
    private WebDriver driver;

    @Before
    public void setUp() {
        System.setProperty("webdriver.chrome.driver", "path/to/chromedriver");
        driver = new ChromeDriver();
        driver.get("https://example.com"); // Replace with the URL of your e-commerce website
    }

    @Test
    public void testAddToCart() {
        // Test case for adding items to the cart
        // Steps:
        // 1. Navigate to the product page.
        // 2. Click on the "Add to Cart" button.
        // 3. Verify that the item is added to the cart.
        WebElement addToCartButton = driver.findElement(By.cssSelector("#add-to-cart-button"));
        addToCartButton.click();
        WebElement cartIcon = driver.findElement(By.cssSelector(".cart-icon"));
        assertTrue(cartIcon.isDisplayed());
    }

    @After
    public void tearDown() {
        driver.quit();
    }
}
